store/storage/pebbledb/db.go

```diff
@@ -0,0 +1,185 @@
+package pebbledb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"cosmossdk.io/store/v2"
+	"cosmossdk.io/store/v2/storage/util"
+	"github.com/cockroachdb/pebble"
+)
+
+const (
+	VersionSize = 8
+
+	StorePrefixTpl   = "s/k:%d/%s/" // s/k:<version>/<storeKey>/<key>
+	latestVersionKey = "s/latest"
+)
+
+var (
+	_ store.VersionedDatabase = (*Database)(nil)
+
+	defaultWriteOpts = pebble.Sync
+)
+
+type Database struct {
+	storage *pebble.DB
+}
+
+func New(dataDir string) (*Database, error) {
+	opts := &pebble.Options{}
+	opts = opts.EnsureDefaults()
```

Comment:
want to be mindful we dont fall into the trap of using the default database settings like we did in the past. It would be good to do some preliminary research on pebble and why the defaults were chosen for the sdk

Reply:
Yes, good point and I totally agree, but note:

1. There is a separate issue to tune/benchmark, i.e. it's not the goal of this PR to get these values right. We just need something out the gate. (ref: https://github.com/cosmos/cosmos-sdk/issues/17322)
2. There is a `NewWithDB` constructor for situations where an app wants to fine tune to it's liking.